### PR TITLE
Feature tmc#127 cs in observers

### PR DESCRIPTION
**Pull request by rphh**

### CODE REVIEW

File: sdl2promela/translator.py [hunk 1]
```diff
@@ -2073,6 +2161,33 @@ def __generate_transition_function(context: Context) -> promelamodel.Inline:
...
 
 
+def __generate_observer_check_inline(context: Context) -> promelamodel.Inline:
+    builder = InlineBuilder()
+    builder.withName(__get_observer_check_function_name(context))
+
+    continuous_signals_present = False
+    for _, signals in context.sdl_model.continuous_signals.items():
+        if len(signals) > 0:
+            continuous_signals_present = True
```
[the remark this answers is not in the document]
Comment (Lurkerpas) [reply sdl2promela/translator.py:2171]: I think that retrieval of this information is done twice, and it is not exactly the same.
 ```
if signal.observer_input is not None:
                    # This is an artificial CS created by OpenGEODE
                    # for an observer.
                    continue
```

File: sdl2promela/translator.py [hunk 2]
```diff
@@ -2103,14 +2218,15 @@ def __generate_aggregate_inline(
     return builder.build()
 
 
-def translate(sdl_model: sdlmodel.Model) -> promelamodel.Model:
+def translate(sdl_model: sdlmodel.Model, observer: bool = False) -> promelamodel.Model:
```
Comment (Lurkerpas) [inline sdl2promela/translator.py:2221]: is_observer?

File: sdl2promela/translator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lurkerpas) [reply sdl2promela/translator.py:2171]: This is after filtering out the fake CS

File: sdl2promela/sdl/model.py [hunk 3]
```diff
@@ -723,6 +723,8 @@ class ObservedSignalKind(Enum):
     OUTPUT = 2
     """Output Observer, inspecting and mutating output from a function."""
 
+    CONTINUOUS_SIGNAL = 3
+
```
[the remark this answers is not in the document]
Comment (rphh) [reply sdl2promela/sdl/model.py:728]: Added

File: sdl2promela/translator.py [hunk 4]
```diff
@@ -1990,12 +2003,87 @@ def __generate_continuous_signals(context: Context) -> promelamodel.Statement:
         AlternativeBuilder().withStatements([promelamodel.Skip()]).build()
     )
 
+    return inner_switch_builder.build()
+
+
+def __generate_continuous_signals_block_for_observer(
+    context: Context,
```
[the remark this answers is not in the document]
Comment (rphh) [reply sdl2promela/translator.py:2013]: New function added, the size of two functions is also reduced significantly

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**